perllib/FixMyStreet/App/Controller/Admin/AreaStats.pm

```diff
@@ -0,0 +1,217 @@
+        } elsif ($area_id and $user->area_id eq $area_id) {
+            return;
+        }
+    }
+
+    $c->detach( '/page_error_404_not_found' );
+}
+
+sub body_base : Chained('/') : PathPart('admin/areastats') : CaptureArgs(1) {
+    my ($self, $c, $body_id) = @_;
+
+    $c->stash->{body_id} = $body_id;
+    my $body = $c->model('DB::Body')->find($body_id);
+    $c->detach( '/page_error_404_not_found', [] )
+      unless $body;
+    $c->stash->{body} = $body;
+    $c->stash->{areas} = mySociety::MaPit::call('area/children', [ $c->stash->{body_id} ] );
+}
+
+sub body_stats : Chained('body_base') : PathPart('') : Args(0) {
+    my ($self, $c) = @_;
+
+    $c->forward('check_user', [$c->stash->{body_id}]);
+    if ($c->get_param('area')) {
+        $c->stash->{area_id} = $c->get_param('area');
+    }
+    $c->forward('stats');
+}
+
+sub ward_stats : Chained('body_base') :PathPart('') : Args(1) {
+    my ($self, $c, $area_id) = @_;
+
+    $c->forward('check_user', [$c->stash->{body_id}, $area_id]);
+
+    $c->stash->{area_id} = $area_id;
+    my $area = mySociety::MaPit::call('area', $area_id );
+    $c->detach( '/page_error_404_not_found' ) if $area->{error};
+    $c->stash->{area} = $area;
+
+    $c->forward('stats');
+}
+
+sub stats : Private {
+    my ($self, $c) = @_;
+
+    my $date = DateTime->now->subtract(days => 30);
+
+    $c->forward('/admin/fetch_contacts');
+
+    $c->stash->{template} = 'admin/areastats/area.html';
+
+    my $dtf = $c->model('DB')->storage->datetime_parser;
+    my $time = $dtf->format_datetime($date);
+
+    my $params = {
+        'me.confirmed' => { '>=', $time },
+    };
+
+    my %area_param = ();
+    if ($c->stash->{area}) {
+        my $area_id = $c->stash->{area_id};
+        $c->stash->{area_name} = $c->stash->{area}->{name};
+        $params->{'problem.areas'} = { like => "%,$area_id,%" };
+        %area_param = (
+            areas => { like => "%,$area_id,%" },
+        );
+    } else {
+        $c->stash->{area_name} = $c->stash->{body}->name;
+    }
+
+    my $comments = $c->model('DB::Comment')->to_body(
+        $c->stash->{body}
+    )->search(
+        $params,
+        {
+            group_by => [ 'problem_state' ],
+            select   => [ 'problem_state', { count => 'me.id' } ],
+            as       => [ qw/state state_count/ ],
+            join     => 'problem'
+        }
+    );
+
+    my $problems = $c->model('DB::Problem')->to_body(
+        $c->stash->{body}
+    )->search(
+        \%area_param,
+        {
+            group_by => [ 'category', 'state' ],
+            select   => [ 'category', 'state', { count => 'me.id' } ],
+            as       => [ qw/category state state_count/ ],
+        }
+    );
+
+    my %by_category = map { $_->category => {} } $c->stash->{contacts}->all;
+    my %recent_by_category = map { $_->category => 0 } $c->stash->{contacts}->all;
+
+    my $state_map = {};
+
+    $state_map->{$_} = 'open' foreach FixMyStreet::DB::Result::Problem->open_states;
```

Comment:
Going by how this is displayed, I think they expect a report to appear only in one of open/scheduled/closed/fixed. So perhaps open should be all open_states that aren't manually being used for scheduled?

Reply:
Isn't that taken care of by:

`$state_map->{$_} = 'scheduled' foreach ('planned', 'action scheduled');`

And for the "problems this month bit" I read that as "n problems opened of which x were scheduled, y were fixed and z closed" so that people can check that problems are flowing through.

Reply:
Oh yes, silly me.